### PR TITLE
Further ideas about Lesson03 and Lesson04

### PR DESCRIPTION
**Pull request by txhsl**
@PeterLinX @merlin217 It's a rough design of a DnD adventure game.

You may have to read the code, because most of the markdown files just have a title. But don't worry, there are some comments to make it easy to understand.

An unknown TypeScript compiling error occurred and it is not likely to be solved in a short time. Another issue to [neo-one](https://github.com/neo-one-suite/neo-one) will be opened.
unknown error

Feel free to comment.

### CODE REVIEW

**Comment by txhsl** [comment]
Here you can find these two issues - [#2016](https://github.com/neo-one-suite/neo-one/issues/2016) and [#2029](https://github.com/neo-one-suite/neo-one/issues/2029).

File: Typescript/en-us/Lesson03/AlienFinder.ts [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+                break;
+            }
+            default: {
+                break;
+            }
+        }
+    }
+
+    private judge(alien: Alien, value: number) {
+        if (value + this.d100() < 200) {
```
Comment (merlin217) [inline Typescript/en-us/Lesson03/AlienFinder.ts:207]: `value` = 0-99, and `this.d100()` = 0-99; so this would be always true, right?

**Comment by merlin217** [comment]
This is brilliant. 
I think the game logic is simple but sufficiently balanced and interesting. I have been trying to implement something similar to `forward()`, but it's not as good as yours. 
I will be implementing these game logic to the C# version in the next few days. 

I also left a comment about `judge()` in the code.

File: Typescript/en-us/Lesson03/AlienFinder.ts (hunk 1 above)
Comment (merlin217) [reply Typescript/en-us/Lesson03/AlienFinder.ts:207]: This event should reward low `weight` in the weight event, and reward high `speed` and `strength` in their events. 

If we use `value + this.d100() < 100`, it rewards low `value`. So for speed and strength event, maybe we can pass in `99 - speed` as the parameter?

Comment (txhsl) [reply Typescript/en-us/Lesson03/AlienFinder.ts:207]: > If value = 0-99, and this.d100() = 0-99;
> then this boolean check would be always true, right?

Well. it must be a bug

**Comment by txhsl** [comment]
@merlin217 Done. 

> If we use value + this.d100() < 100, it rewards low value. So for speed and strength event, maybe we can pass in 99 - speed as the parameter?

This will be considered in the future.

**Pull request merged**